pkgs/by-name/br/breakpointHook/attach.sh

```diff
@@ -0,0 +1,28 @@
+#! /usr/bin/env bash
+
+id="$1"
+pid="$(pgrep -f "sleep $id" | head -n1)"
+
+# helper to extract variables from the build env
+getVar(){
+  nsenter --mount --target "$pid" /bin/sh -c "
+    while read line; do
+      case \$line in
+        *\"declare -x $1=\"* )
+          echo \$line
+            ;;
+      esac
+    done < /build/env-vars
+  " | cut -d \" -f 2
+}
+
+# the build shell is needed to `source /build/env-vars``
+buildShell="$(getVar shell)"
+# the debug shell will be started as interactive shell after sourcing /build/env-vars
+debugShell="$(getVar debugShell)"
+# to drop the user into the working directory at the point of failure
+pwd="$(getVar PWD)"
+
+# enter the namespace of the failed build
+exec nsenter --mount --target "$pid" "$buildShell" -c \
+  "source /build/env-vars && cd $pwd && exec $debugShell"
```

Comment:
Maybe we can pass in `/proc/$pid/environ` as fd 3:

```suggestion
  "while read ...; do export ...; done <3 && cd $pwd && exec $debugShell" <3 /proc/$pid/environ
```

Reply:
what about shell functions? they would not be sourced

Reply:
Oh they are already not included

Reply:
OK. I implemented it using read from fd 3